react/javascript/demo-embeds/src/components/DashboardEmbedNext/DashboardEmbedNext.jsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2021 Looker Data Sciences, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import React, { useCallback, useContext, useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import {
+  Button,
+  Layout,
+  Page,
+  Aside,
+  Section,
+  MessageBar,
+  Box,
+  SpaceVertical,
+} from '@looker/components'
+import { ExtensionContext2 } from '@looker/extension-sdk-react'
+import { LookerEmbedSDK } from '@looker/embed-sdk'
+import { useSearchDashboards } from '../../hooks/use_search_dashboards'
+import { useCurrentRoute, useNavigate, useListenEmbedEvents } from '../../hooks'
+import { Search } from '../Search'
+import { EmbedContainer } from '../EmbedContainer'
+import { EmbedEvents } from '../EmbedEvents'
+
+export const DashboardEmbedNext = ({ embedType }) => {
+  const { searchCriteria, embedId } = useCurrentRoute(embedType)
+  const { updateSearchCriteria, updateEmbedId } = useNavigate(embedType)
+  const { extensionSDK } = useContext(ExtensionContext2)
+  const [criteria, setCriteria] = useState(searchCriteria || '')
+  const [message, setMessage] = useState()
+  const [running, setRunning] = useState(true)
+  const [dashboardId, setDashboardId] = useState()
+  const [dashboard, setDashboard] = useState()
+  const { data, isLoading, error } = useSearchDashboards(criteria, embedType)
+  const results = (data || []).map(({ id, title }) => ({
+    id,
+    description: title,
+  }))
+  const { embedEvents, listenEmbedEvents } = useListenEmbedEvents()
+
+  useEffect(() => {
+    if (dashboardId && dashboardId !== embedId) {
+      if (dashboard) {
```

Comment:
Looks like you could have combined all 3 conditions here :) this is fine as is though

Reply:
You right. There used to be an else but that caused problems with page reloads so I removed it.

I'm going to leave as is for now but will probably fix in the next PR.